cli/src/main/java/com/devonfw/tools/ide/cli/IdeCompleter.java

```diff
@@ -0,0 +1,201 @@
+package com.devonfw.tools.ide.cli;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jline.reader.Candidate;
+import org.jline.reader.Completer;
+import org.jline.reader.LineReader;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.impl.completer.ArgumentCompleter;
+import org.jline.reader.impl.completer.NullCompleter;
+import org.jline.utils.AttributedString;
+
+import com.devonfw.tools.ide.commandlet.Commandlet;
+import com.devonfw.tools.ide.commandlet.ContextCommandlet;
+import com.devonfw.tools.ide.commandlet.HelpCommandlet;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.property.FlagProperty;
+import com.devonfw.tools.ide.property.LocaleProperty;
+import com.devonfw.tools.ide.property.Property;
+import com.devonfw.tools.ide.property.ToolProperty;
+import com.devonfw.tools.ide.property.VersionProperty;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+/**
+ * Implements the {@link Completer} for jline3 autocompletion. Inspired by picocli
+ */
+public class IdeCompleter extends ArgumentCompleter implements Completer {
+
+  /** Pattern which should stop autocompletion e.g. 'get-version mvn -' (prevents options after tool commands) */
+  private static final String INVALID_PATTERN = "([\\s][a-z]*[\\s][\\-](\\s)*.*$)|([a-z][\\-][a-z]*[\\s][\\-](\\s)*.*$)|(^[a-z]*(\\s)[\\-](\\s)*.*$)";
```

Comment:
Can you pls add a test for this regex (with words that contain "-" in middle of the word, e.g. "foo-bar")? Maybe change [a-z] to \w or \S to make this regex also work when the words contain chars like [A-Z], _, üöüÜÖÄ

Reply:
I've added another test which checks these cases now.
Further checks for more cases should be added to a new issue I think.